src/cephadm/cephadm

```diff
@@ -322,10 +322,126 @@ class NFSGanesha(object):
 
 ##################################
 
+class CephIscsi(object):
+    """Defines a Ceph-Iscsi container"""
+
+    daemon_type = 'iscsi'
+    entrypoint = '/usr/bin/rbd-target-api'
+
+    required_files = ['iscsi-gateway.cfg']
+
+    def __init__(self,
+                 fsid,
+                 daemon_id,
+                 config_json,
+                 image=DEFAULT_IMAGE):
+        # type: (str, Union[int, str], Dict, str) -> None
+        self.fsid = fsid
+        self.daemon_id = daemon_id
+        self.image = image
+
+        def json_get(key, default=None, require=False):
+            if require and not key in config_json.keys():
+                raise Error('{} missing from config-json'.format(key))
+            return config_json.get(key, default)
+
+        # config-json options
+        self.files = json_get('files', {})
+
+        # validate the supplied args
+        self.validate()
+
+    @classmethod
+    def init(cls, fsid, daemon_id):
+        # type: (str, Union[int, str]) -> CephIscsi
+        return cls(fsid, daemon_id, get_parm(args.config_json), args.image)
+
+    @staticmethod
+    def get_container_mounts(data_dir, log_dir):
+        # type: (str, str) -> Dict[str, str]
+        mounts = dict()
+        mounts[os.path.join(data_dir, 'config')] = '/etc/ceph/ceph.conf:z'
+        mounts[os.path.join(data_dir, 'keyring')] = '/etc/ceph/keyring:z'
+        mounts[os.path.join(data_dir, 'iscsi-gateway.cfg')] = '/etc/ceph/iscsi-gateway.cfg:z'
+        mounts[log_dir] = '/var/log/rbd-target-api:z'
+        # Sys below is for configfs. Really should only need
+        # /sys/kernel/config but that produces a permission denied,
+        # when trying to bind mount with docker (needs more
+        # investigation)
+        mounts['/sys'] = '/sys'
```

Comment:
this makes me nervous... maybe mount this as read-only ?

 ```suggestion
         # Sys below is for configfs. Really should only need
         # /sys/kernel/config but that produces a permission denied,
         # when trying to bind mount with docker (needs more
         # investigation)
         mounts['/sys'] = '/sys:ro'
```

Reply:
OK been playing with this. Turns out we can mount /sys/kernel/config, but only if the configfs has already been mounted. Which it doesn't do dy default.

Next I went a step further and changed the mount to: 

  `mounts[os.path.join(data_dir, 'configfs')] = '/sys/kernel/config:z'`

Which if I mount manually with:

  `mount -t configfs none /var/lib/ceph/<fsid>/<daemon>/configfs`

works.

So next, I've gone an added a `mount` method to cephadm that really just calls one of the `call` methods. But doesn't work, gives the error:
`mount: /var/lib/ceph/3e264f40-e716-4198-bf1b-dca8e6e15e7b/iscsi.iscsi.ironic-moliver.tdtzdz/configfs: unknown filesystem type ' configfs'.`

I guess it makes sense, I guess it is a security issue. But we also can't expect the user to mount it or start targetcli on the potential nodes before we deploy.
The main problem is, obviously, LIO isn't running before we deploy on nodes so configfs hasn't been mounted yet.